Zend/zend_vm_def.h

```diff
@@ -4074,6 +4082,10 @@ ZEND_VM_HOT_HANDLER(60, ZEND_DO_FCALL, ANY, ANY, SPEC(RETVAL,OBSERVER))
 			ret = EX_VAR(opline->result.var);
 		}
 
+		if (UNEXPECTED((fbc->common.fn_flags & ZEND_ACC_DEPRECATED) != 0)) {
+			zend_deprecated_function(fbc);
```

Comment:
Yes, i haven't copied this over because i don't fully understand it yet, but left open a todo. A test that converts deprecation into exception will probably reveal what this is used for.